src/components/ColumnNameItem/ColumnNameItem.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+
+import { SearchLink } from '../SearchLink';
+import { ColumnNames } from '../../types';
+
+type Props = {
+  value: keyof typeof ColumnNames,
+};
+
+export const ColumnName: React.FC<Props> = ({ value }) => {
+  const [searchParams] = useSearchParams();
+
+  const lowerCasedValue = value.toLowerCase();
+
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const getSortParams = (sortOption: string) => {
+    if (sortOption !== sort) {
+      return { sort: sortOption, order: null };
+    }
+
+    if (sort === sortOption && !order) {
+      return { sort: sortOption, order: 'desc' };
+    }
+
+    return { sort: null, order: null };
+  };
```

Comment:
It is better to define this function outside the component so that it is not created every time a new render is performed.